lib/logflare/backends/adaptor/tcp_adaptor/syslog.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Logflare.Backends.Adaptor.TCPAdaptor.Syslog do
+  @moduledoc """
+  Implementation of [RFC5424 The Syslog Protocol][]
+
+  It uses [octet-counting framing][].
+
+  [RFC5424]: https://www.rfc-editor.org/rfc/rfc5424
+  [octet-counting framing]: https://www.rfc-editor.org/rfc/rfc6587#section-3.4.1
+  """
+
+  alias Logflare.LogEvent
+
+  # TODO: Change it to real value
+  @pen 62137
+
+  def format(%LogEvent{} = le, options) do
+    msg = [
+      header(le, options),
+      " ",
+      structured_data(le, options),
+      " ",
+      Jason.encode!(le.body),
+      "\n"
+    ]
+
+    # TODO: Add support for non-transparent framing
+    [to_string(IO.iodata_length(msg)), ?\s, msg]
+  end
+
+  defp header(%LogEvent{} = le, options) do
+    level = to_level(le.body["level"] || le.body["metadata"]["level"])
+    facility = options[:facility] || 16
+
+    ingested_at = DateTime.from_naive!(le.ingested_at, "Etc/UTC")
+
+    id = Ecto.UUID.dump!(le.id) |> Base.encode32(case: :lower, padding: false)
```

Comment:
Because of 32 characters limit I decided to encode UUIDs with Base32. Theoretically we could just remove `-` (dashes) from canonical representation, but I think that this approach is effective enough as well as it reduces amount of data that needs to be transferred.